### PR TITLE
Update the return type of `convert_qubo_to_model` part 2

### PR DESCRIPTION
**Pull request by vtomole**
Fixes https://github.com/SupertechLabs/general-superstaq/issues/29

### CODE REVIEW

File: general_superstaq/qubo.py [hunk 1]
```diff
@@ -1,10 +1,13 @@
-from typing import Dict, Iterable, List
+from typing import TYPE_CHECKING, Dict, List
 
 import numpy as np
 import qubovert as qv
 
 import general_superstaq as gss
 
+if TYPE_CHECKING:
```
Comment (richrines1) [inline general_superstaq/qubo.py:8]: at least locally everything seems to pass without this `if`, if you also remove the quotes from "QuboDict" below

Comment (vtomole) [reply general_superstaq/qubo.py]: Wasn't explicitly importing the typing QubotDict. Fixed.

**Comment by richrines1** [comment]
i think the "qubo" entries in the other gss.typing TypedDicts should also be changed to `List[QuboDict]`

**Review by perlinm — APPROVED** [review]
lgtm

**Comment by richrines1** [comment]
> lgtm

ditto

**Pull request merged**